src/components/ui/text-field.tsx

```diff
@@ -32,131 +40,150 @@ type Props = React.ComponentProps<typeof TextInput> & {
   size?: 'small' | 'large';
 };
 
-export const TextField: React.FC<Props> = ({
-  size = 'small',
-  label,
-  error,
-  errorText,
-  value,
-  style,
-  onBlur,
-  onFocus,
-  placeholder,
-  rightAction,
-  multiline,
-  ...restOfProps
-}) => {
-  const isLarge = size === 'large';
-  const {width} = useWindowDimensions();
-
-  const [isFocused, setIsFocused] = useState(false);
-  const [height, setHeight] = useState(0);
-
-  const inputRef = useRef<TextInput>(null);
-  const focusAnim = useRef(new Animated.Value(0)).current;
-
-  useEffect(() => {
-    Animated.timing(focusAnim, {
-      toValue: isFocused || !!value ? 1 : 0,
-      duration: 150,
-      easing: Easing.bezier(0.4, 0, 0.2, 1),
-      useNativeDriver: true,
-    }).start();
-  }, [focusAnim, isFocused, value]);
-
-  let color = LIGHT_TEXT_BASE_2;
-  if (error) {
-    color = LIGHT_TEXT_RED_1;
-  }
-
-  const top = -4;
-  const getHeight = height + 30;
-
-  return (
-    <>
-      <View
-        style={[
-          page.container,
-          style,
-          error && page.containerError,
-          {height: getHeight},
-        ]}>
-        <TextInput
-          selectionColor={LIGHT_TEXT_GREEN_1}
-          allowFontScaling={false}
+const AnimatedTextInput = Animated.createAnimatedComponent(TextInput);
+type sizeChangeEventType =
+  NativeSyntheticEvent<TextInputContentSizeChangeEventData>;
+type nativeEventType = sizeChangeEventType['nativeEvent'];
+
+export const TextField: React.FC<Props> = memo(
+  ({
+    size = 'small',
+    label,
+    error,
+    errorText,
+    value,
+    style,
+    onBlur,
+    onFocus,
+    placeholder,
+    rightAction,
+    multiline,
+    ...restOfProps
+  }) => {
+    const isLarge = size === 'large';
+    const {width} = useWindowDimensions();
+    const [isFocused, setIsFocused] = useState(false);
+
+    const inputRef = useRef<TextInput>(null);
+    const height = useSharedValue(30);
+    const focusAnim = useSharedValue(0);
+
+    const onChangeContentSize = (newHeight: number) => {
+      'worklet';
+      const inputH = Math.max(newHeight, 28);
+      height.value = inputH + 30;
+    };
+
+    const contentSizeChangeEvent = useEvent((event: nativeEventType) => {
+      'worklet';
+      onChangeContentSize(event.contentSize.height);
+    });
+
+    const contentSizeChangeEventIOS = (event: sizeChangeEventType) => {
+      'worklet';
+      onChangeContentSize(event.nativeEvent.contentSize.height);
+    };
+
+    const onInputEvent = useCallback(() => {
+      focusAnim.value = withTiming(isFocused || !!value ? 1 : 0, {
+        duration: 150,
+        easing: Easing.bezier(0.4, 0, 0.2, 1),
+      });
+    }, [value, focusAnim, isFocused]);
+
+    useEffect(() => {
+      onInputEvent();
+    }, [onInputEvent]);
```

Comment:
Is there any reason why it's just put this callback into single useEffect like this?
```ts
useEffect(() => {
  focusAnim.value = withTiming(isFocused || !!value ? 1 : 0, {
    duration: 150,
    easing: Easing.bezier(0.4, 0, 0.2, 1),
  });
}, [value, focusAnim, isFocused]);
```

Reply:
This will work analogically, because if `[value, focusAnim, isFocused]` change, then the reference to the `onInputEvent` function will change, which will cause `useEffect`
The advantage of this change is that now the function can be called not only in `useEffect`

Reply:
but these code called only in useEffect =)